### PR TITLE
fix(mrm_handler): add check for autonomous mode and do some refactoring

### PR DESCRIPTION
**Pull request by danielsanchezaran**
## Description
Currently, if use_emergency_holding is set to true, the ego will enter emergency mode if it is not changed to autonomous mode within "timeout_emergency_recovery" seconds, which is not the expected behavior. 

This PR adds a check for Autonomous mode before increasing the timer, which solves the issue. 

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
PSim

https://github.com/user-attachments/assets/f5441097-dd87-41ba-81b6-93d6faa81668

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp [hunk 1]
```diff
@@ -82,28 +82,26 @@ void MrmHandler::onOperationModeAvailability(
...
-        }
-      }
-    }
+  if (msg->autonomous) {
+    stamp_autonomous_become_unavailable_.reset();
+    return;
   }
-  operation_mode_availability_ = msg;
+
+  stamp_autonomous_become_unavailable_ = (!stamp_autonomous_become_unavailable_.has_value())
```
Comment (veqcc) [inline system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:98]: I prefer the following. What do you think?

```
if (stamp_autonomous_become_unavailable_.has_value()) {
  const auto emergency_duration =
    (this->now() - stamp_autonomous_become_unavailable_.value()).seconds();
  is_emergency_holding_ = (emergency_duration > param_.timeout_emergency_recovery);
} else {
  stamp_autonomous_become_unavailable_.emplace(this->now());
}

**Comment by TetsuKawa** [comment]
LGTM If you correct Kambe's points.

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp (hunk 1 above)
Comment (danielsanchezaran) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:98]: @veqcc I don't think it is worth it to add branching to the code, it decreases readability and increases cyclomatic complexity. Adding the else branch only skips these checks: 

`  const auto emergency_duration =
    (this->now() - stamp_autonomous_become_unavailable_.value()).seconds();
  is_emergency_holding_ = (emergency_duration > param_.timeout_emergency_recovery);`

which are not costly to compute. Also the cost of computing these operations might be lower than having a branching code. I believe Adding comments is ok, but dont agree with using an else branch if it is not necessary.

Comment (veqcc) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:98]: Agreed :+1:

**Review by veqcc — APPROVED** [review]

**Pull request merged**